libspectrum/doc/libspectrum.3
.\" -*- nroff -*-
.\"
.\" libspectrum.3: libspectrum summary
.\" Copyright (c) 2004-2013 Philip Kendall
.\"
.\" This program is free software; you can redistribute it and/or modify
.\" it under the terms of the GNU General Public License as published by
.\" the Free Software Foundation; either version 2 of the License, or
.\" (at your option) any later version.
.\"
.\" This program is distributed in the hope that it will be useful,
.\" but WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\" You should have received a copy of the GNU General Public License along
.\" with this program; if not, write to the Free Software Foundation, Inc.,
.\" 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
.\"
.\" Author contact information:
.\"
.\" E-mail: philip-fuse@shadowmagic.org.uk
.\"
.\"
.TH libspectrum 3 "21st August, 2016" "Version 1.2.2" "Emulators"
.\"
.\"------------------------------------------------------------------
.\"
.SH NAME
libspectrum \(em Sinclair ZX Spectrum support library
.\"
.\"------------------------------------------------------------------
.\"
.SH DESCRIPTION
.I libspectrum
is a library for reading and writing some of the file formats used by
emulators of the Sinclair ZX Spectrum. It also contains routines
providing information about the various Spectrum clones which were
available.
.PP
The file formats currently supported by
.I libspectrum
are:
.IP \(bu 2
Snapshots: .z80, .szx, .sna (all read/write), .zxs, .sp., .snp and +D
snapshots (read only).
.IP \(bu 2
Tape images: .tzx, .tap, .spc, .sta, .ltp (read/write) and .pzx, Warajevo .tap,
Z80Em and CSW version 1 (read only).
.IP \(bu 2
Input recordings: .rzx (read/write).
.IP \(bu 2
Disk images: .dsk (both plain and extended), .d40, .d80, .fdi, .img, .mgt,
\[char46]opd, .sad, .scl, .td0, .trd and .udi (identification only).
.IP \(bu 2
Timex cartridges: .dck (read only).
.IP \(bu 2
IDE hard disk images: .hdf (read/write).
.IP \(bu 2
Microdrive cartridge images: .mdr (read/write).
.PP
Additionally, any files compressed with
.IR bzip2 "(3),"
.IR gzip "(3)"
or
.IR zip "(3)"
can be read and will be transparently decompressed.
.PP
Full documentation for
.I libspectrum
is provided in
.RI ` libspectrum.txt '.
.\"
.\"------------------------------------------------------------------
.\"
.SH SEE ALSO
.IR bzip2 "(3),"
.IR fuse "(1),"
.IR fuse\-utils "(1),"
.IR gzip "(3),"
.IR zip "(3)."
.PP
The comp.sys.sinclair Spectrum FAQ, at
.br
.IR "http://www.worldofspectrum.org/faq/index.html" .
.\"
.\"------------------------------------------------------------------
.\"
.SH AUTHOR
Philip Kendall (philip\-fuse@shadowmagic.org.uk)


fuse/ChangeLog
2016-07-17  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * Fuse 1.2.1 released.

        * Emulation core improvements:
          * Fix bugs when the detect loaders feature is being used (thanks, Lee
            Tonks and windale) (Fredrick Meunier).

        * Debugger improvements:
          * Remove the need for "%" when accessing system variables (Philip
            Kendall).
          * Add Z80 registers as debugger variables (Philip Kendall).
          * Expose last byte written to the ULA, tstates since interrupt,
            primary and secondary memory control ports as debugger system
            variables (Philip Kendall).
          * Make breakpoints on events honour lifetime (Sergio Baldoví).
          * Extend breakpoints on paging events to more peripherals: Beta 128,
            +D, Didaktik 80, DISCiPLE, Opus Discovery and SpeccyBoot (Sergio
            Baldoví).
          * Split +D memory sources into RAM and ROM sections (Sergio Baldoví).
          * Coalesce +D and DISCiPLE RAM pages so they show as 8K pages (Sergio
            Baldoví).

        * Miscellaneous improvements:
          * Update GNOME .desktop file (Alberto Garcia).
          * Add an emulator module startup manager to automatically handle
            dependency issues (Philip Kendall).
          * Fix crash on widget UIs when hitting the close icon on the title bar
            several times (Sergio Baldoví).

2016-06-06  Philip Kendall  <philip-fuse@shadowmagic.org.uk>

        * Fuse 1.2.0 released.

        * New features:
          * Add Currah µSource emulation (Stuart Brady).
          * Add Didaktik 80/40 emulation (Gergely Szasz).
          * Capture BASIC video functions to SVG file (Stefano Bodrato).
          * Support bash completion (Sergio Baldoví).

        * Emulation core improvements:
          * Allow continuing RZX recordings if there is a final snapshot in the
            RZX (Sergio Baldoví).
          * Fix the prune function on RZX rollback (Sergio Baldoví).
          * Use SZX format for the initial snapshot in RZX files (Sergio
            Baldoví).
          * Fix loading of EDSK files with Sector Offset block (Sergio Baldoví).
          * Migrate disk "index event" handling to the FDD layer and have the
            FDC layer use it for their STATUS registers (Gergely Szasz).
          * Implement WD2797 emulation (Gergely Szasz).
          * Centralise the "Disk icon" update code to the FDD layer (Gergely
            Szasz).
          * Fix disk image corruption after saving UDI files (Sergio Baldoví).
          * Check ready status after loading a disk into floppy disk drive
            (thanks, John Elliott) (Sergio Baldoví).
          * Fix overlapped SEEK commands (Sergio Baldoví).
          * Fix length of data returned by READ_DIAG (thanks, Fredrick Meunier)
            (Sergio Baldoví).
          * Fix writing .td0 format disk files (Sergio Baldoví).
          * Fix speech in Cobra's Arc - Medium Case.tzx when loaded with tape
            traps enabled (thanks, zx81 and Sergio Baldoví) (Fredrick Meunier).
          * Skip tape traps if VERIFY is requested (UB880D).
          * Fix loading sound with some custom loaders (Fredrick Meunier).
          * Check if data blocks are headers for handling PZX files (Fredrick
            Meunier).
          * Correct display of pulses in PZX pulse block for tape browser
            (Fredrick Meunier).
          * Set AF, AF' and SP to 0xffff on reset (Stuart Brady).
          * Leave most registers unchanged on a soft reset (Stuart Brady).
          * Emulate interrupt and NMI timings more precisely (Stuart Brady).
          * Emulate NMOS and CMOS Z80 variants (Stuart Brady).
          * Switch to 2KB page size (Stuart Brady).
          * Fix inaccurate output when AY envelopes are used (Matthew Westcott
            and Fredrick Meunier).
          * Fix some peripherals activation when loading snapshots (Sergio
            Baldoví).

        * Machine specific improvements:
          * The address range 0x4000 - 0x7FFF is contended on the TS2068 in the
            home, Dock and Exrom banks (thanks, Richard Atkinson). It is assumed
            that this is the same for other Timex models (Fredrick Meunier).
          * Clear all Opus RAM on hard reset (Stuart Brady).
          * Ensure the ZX Printer does not require a hard reset to enable
            (thanks, RMartins) (Fredrick Meunier).
          * Fix timing of events when emulating Scorpion (Stuart Brady).
          * Ensure we have successfully selected a Pentagon or a Timex machine
            before inserting their media (Fredrick Meunier).
          * Prefer Scorpion to Pentagon when loading SCL/TRD disks for better
            timing compatibility (thanks, windale) (Sergio Baldoví).
          * Fix Beta 128 type II commands (thanks, windale and BogDan Vatra)
            (Sergio Baldoví).
          * Don't enable the Beta 128 interface when loading a snapshot on a
            machine with Beta built-in (thanks, windale and BogDan Vatra)
            (Fredrick Meunier).
          * Lock port +3 1FFDh if paging is disabled (Brian Ruthven and Fredrick
            Meunier).
          * Spectranet: fix segfault in error handling when setting SO_REUSEADDR
            (Stuart Brady).
          * Update SE ROMs to v4.07 (thanks, Andrew Owen) (Sergio Baldoví).

        * Debugger improvements:
          * Fix disassembly of LD (HL), LD (IX) and LD (IY) (BogDan Vatra).
          * Add I and R register setting and getting to the debugger (Sergio
            Baldoví).
          * Show the status of the halted flag in the debugger (Stuart Brady).
          * Fix time breakpoints later than a frame in the future (Sergio
            Baldoví).
          * Timex EXROM and Dock text was truncated in the debugger UI (thanks,
            Andrew Owen and Sergio Baldoví) (Fredrick Meunier).
          * Win32: relocate halted flag and set monospaced font (Sergio

website/libspectrum.php
<?php

/* libspectrum.php: the libspectrum 'home' page
   Copyright (c) 1999-2010 Darren Salt, Philip Kendall

   $Id$

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 2 of the License, or
   (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License along
   with this program; if not, write to the Free Software Foundation, Inc.,
   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

   Author contact information:

   E-mail: philip-fuse@shadowmagic.org.uk

*/

include ".inc/fuse.inc";

fuse_title ("libspectrum - emulator support library");
fuse_menu_heading ("libspectrum");

#<!-- ======================================================= -->

fuse_section (NULL, "What is it?", <<<END_SECTION
  <p>libspectrum is a library designed to make the input and output of some ZX Spectrum emulator files slightly easier. It is intended to be usable on Unix variants, Win32 and Mac OS X. Currently, it is mainly (only?) used by <a href="fuse.php">Fuse</a>, but other Spectrum emulator and utility authors are encouraged to use it.</p>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section ("Supported formats", "Which formats does it support?", <<<END_SECTION
  <ul>
   <li>Snapshots:
    <ul>
     <li>read/write: .z80, .szx, .sna</li>
     <li>read only: .zxs, .sp., .snp, +D snapshots</li>
    </ul>
   </li>
   <li>Tape images:
    <ul>
     <li>read/write: .tzx, .tap, .spc, .sta, .ltp</li>
     <li>read only: .pzx, Warajevo .tap, Z80Em, CSW version 1, .wav</li>
    </ul>
   </li>
   <li>Input recordings:
    <ul><li>read/write: .rzx</li></ul>
   </li>
   <li>Timex cartridges:
    <ul><li>read only: .dck</li></ul>
   </li>
   <li>IDE hard disk images:
    <ul><li>read/write: .hdf</li></ul>
   </li>
   <li>Microdrive cartridge images:
    <ul><li>read/write: .mdr</li></ul>
   </li>
   <li>Compressed files: reading of gzipped and bzipped files is transparently supported.</li>
  </ul>

  <p>There are also some useful utility routines to do things like identify Spectrum files, or give the timings for various machines. For more details, see the <a href="libspectrum.txt">supplied documentation</a>.</p>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section ("Download", "Downloads", <<<END_SECTION

  <p>libspectrum is licensed under the <a href="COPYING">GNU General Public License</a>, version 2 or later. Please read this before downloading libspectrum if you're not already familiar with it.</p>

  <h3>Source</h3>
  <ul>
   <li><a href="https://sourceforge.net/projects/fuse-emulator/files/libspectrum/1.2.1/libspectrum-1.2.1.tar.gz/download">libspectrum-1.2.1.tar.gz</a> (<a href="https://sourceforge.net/projects/fuse-emulator/files/libspectrum/1.2.1/libspectrum-1.2.1.tar.gz.sig/download">PGP signature</a>)<!-- or from <a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/libspectrum-1.1.1.tar.gz">World of Spectrum</a> (<a href="ftp://ftp.worldofspectrum.org/pub/sinclair/emulators/unix/libspectrum-1.1.1.tar.gz.asc">signature</a>)-->.</li>
   <li>The source code release above is signed with the <a href="http://wwwkeys.pgp.net:11371/pks/lookup?op=get&amp;search=0xD0767AB9">Fuse Release Key, ID <tt>D0767AB9</tt></a>, which has fingerprint <tt>064E 0BA9 688F 2699 3267 B1E5 1043 EEEB D076 7AB9</tt>. This is different from the key used to sign the 0.1.1 release as I forgot the passphrase for that key <tt>:-(</tt>.</li>
  </ul>

  <h3>Binaries</h3>
  <p>Packages are available for some Unix distributions; in general, any problems which are specific to the packages should be sent to the package maintainer.</p>
  <ul>
   <li><a href="https://www.archlinux.org/">Arch Linux</a> has 1.1.1 available from the <a href="https://aur.archlinux.org/packages/libspectrum/">Arch User Repository</a>.</li>
   <li><a href="http://packages.debian.org/search?keywords=libspectrum">Official packages</a> of 1.1.1 for <a href="http://www.debian.org/">Debian</a>, maintained by Alberto Garcia, are available.</li>
   <li><a href="https://apps.fedoraproject.org/packages/libspectrum">Fedora</a> (1.1.1)</li>
   <li><a href="http://svnweb.freebsd.org/ports/head/emulators/libspectrum/">FreeBSD</a> (1.1.1)</li>
   <li><a href="https://packages.gentoo.org/packages/app-emulation/libspectrum">Gentoo</a> (1.1.1)</li>
   <li>Mandriva packages of 0.5.0.1 were available from the <a href="http://plf.zarb.org/">PLF</a>, or some <a href="http://sinclairmania.speccy.org/arch/emu/mandrake/">older packages</a> from Miguel Barrio Orsikowsky.</li>
   <li><a href="http://www.netbsd.org/">NetBSD</a> has 1.1.1 available as <a href="http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/emulators/libspectrum/">emulators/libspectrum</a> in <a href="http://www.pkgsrc.org/">pkgsrc</a>.</li>
   <li><a href="http://www.openbsd.org/">OpenBSD</a> has 1.1.1 available as libspectrum package.</li>
   <li><a href="https://build.opensuse.org/package/show/Emulators/libspectrum">openSUSE</a> (1.1.1)</li>
   <li><a href="http://www.slackware.com/">Slackware</a> has 1.1.1 available from <a href="http://slackbuilds.org/result/?search=libspectrum">SlackBuilds</a>.</li>
   <li><a href="https://launchpad.net/ubuntu/+source/libspectrum">Ubuntu</a> (1.1.1)</li>
  </ul>

END_SECTION
);

#<!-- ======================================================= -->

fuse_section (NULL, "Development", <<<END_SECTION
  <p>If you're interested in working on libspectrum, subscribe to the <a href="https://lists.sourceforge.net/lists/listinfo/fuse-emulator-devel">fuse-emulator-devel</a> mailing list and/or see the <a href="https://sourceforge.net/projects/fuse-emulator/">project page</a> on SourceForge. If you just want news of new versions, you probably want the (low volume) <a href="https://lists.sourceforge.net/lists/listinfo/fuse-emulator-announce">fuse-emulator-announce</a> list instead.</p>

  <p>Finally, the latest development version should always be available as the 'trunk/libspectrum' directory from the <a href="https://sourceforge.net/p/fuse-emulator/code/">Subversion repository</a> on SourceForge. Don't expect any support for this, or even that it can be compiled...</p>

END_SECTION
);

#<!-- ======================================================= -->

fuse_related ("Fuse", "fuse.php");

fuse_footer ();
?>
